### DISCUSSION TITLE
Trying to compare date with datetime

### DISCUSSION THREAD

**Discussion by leonardopottmayer** [prisma/prisma#20219] [Q&A] [answered]
Hello. Well, i'm creating a functions wich reads the query from express and transforms it into prisma filters, like with where, in, gte, etc. It's working fine, but when i try to compare some datetimes, it doesn't work.

For example:
In my database i have a transaction_Category table, wich has a createdAt column (defined as DateTime). So the value on it is something like `"2023-06-25 17:12:16.000"`.

The object i'm sending to the filter is similar to this:
`{ createdAt: { in: [ 2023-06-25T00:00:00.000Z ] } }`
or
`{ createdAt: { lte: 2023-06-25T00:00:00.000Z } }`

The idea is to filter date without comparing the time, so i can do like 
`{ createdAt: { gte: 2023-06-25T00:00:00.000Z, lte: gte: 2023-06-26T00:00:00.000Z } }`

Can someone please help me for solving this issue? I've already tried to add a new column to my model, called truncCreatedAtm wich is a @db.Date column and just stores the date. But it didn't help me.

Thanks for any reply or help :)

**Comment by ludralph** [comment]
Hi @leonardopottmayer 👋 

Thank you for raising this question.

To filter by date without comparing the time, you can define your schema like this:

```prisma
model Post {
  id Int @id @default(autoincrement())
  name String
  createdAt DateTime @default(now()) @db.Date
}
```

You can write the query like this 
```js
 const res = await prisma.post.findMany({
    where:{
      createdAt:{
        lte: new Date('2023-07-12'),
        gte: new Date('2023-07-11')
      }
    }
  })
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by leonardopottmayer** [reply, marked as answer]
Hey, thanks! Yes it works. But what should i do if i want to compare one single date, like 

prisma.post.findMany({ where: { createdAt: 2023-07-12 } }). 

I found another solution, by receiving the date and setting date utc to 0, 0, 0, 0 (gte operator) and creating a new date and setting the utc to 23, 59, 59, 999 (lte operator).

I'd also like to ask if there is any library to convert query search parameters to prisma filters. I'm coding my own, so i can usa lte, lt, gte, gt, in, notIn, equals, and contains. But it's not that easy XD.

**Reply by ludralph** [reply]
I'm not aware of any such library at the moment.
